workflow/artifacts/oss/oss.go

```diff
@@ -59,21 +62,25 @@ func (ossDriver *ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string)
 				return !isTransientOSSErr(err), err
 			}
 			objectName := inputArtifact.OSS.Key
-			origErr := bucket.GetObjectToFile(objectName, path)
-			if origErr == nil {
+			err = bucket.GetObjectToFile(objectName, path)
+			if err == nil {
 				return true, nil
 			}
-			if ossErr, ok := origErr.(oss.ServiceError); ok {
-				if ossErr.Code != "NoSuchKey" {
-					return !isTransientOSSErr(err), fmt.Errorf("failed to get file: %w", origErr)
-				}
+			if !IsOssErrCode(err, "NoSuchKey") {
+				return !isTransientOSSErr(err), fmt.Errorf("failed to get file: %w", err)
 			}
-			// If we get here, the error was a NoSuchKey. The key might be a directory.
-			// There is only one method in OSS for downloading objects that does not differentiate between a file
-			// and a directory so we append the a trailing slash here to differentiate that prior to downloading.
-			err = bucket.GetObjectToFile(objectName+"/", path)
+			// If we get here, the error was a NoSuchKey. The key might be a oss "directory"
+			isDir, err := IsOssDirectory(bucket, objectName)
 			if err != nil {
-				return !isTransientOSSErr(err), err
+				return !isTransientOSSErr(err), fmt.Errorf("failed to test if %s/%s is a directory: %w", bucketName, objectName, err)
+			}
+			if !isDir {
+				// It's neither a file, nor a directory. Return the original NoSuchKey error
+				return false, errors.New(errors.CodeNotFound, err.Error())
```

Comment:
This `err` is overwritten as part of the `IsOssDirectory` check though so the original error is masked out

Reply:
You are right, I'll fix it.